### PR TITLE
split room and participant services

### PR DESCRIPTION
**Pull request by paulwe**
- inject psrpc clients with wire
- move config to protocol to fix dependencies
- use multiton service helper to handle room/participant deregistration

### CODE REVIEW

File: pkg/service/roommanager.go [hunk 1]
```diff
@@ -433,13 +435,17 @@ func (r *RoomManager) StartSession(
 		_ = participant.Close(true, types.ParticipantCloseReasonJoinFailed, false)
 		return err
 	}
-	if r.config.PSRPC.Enabled {
-		if err := r.roomServer.RegisterAllParticipantTopics(rpc.FormatParticipantTopic(roomName, participant.Identity())); err != nil {
-			pLogger.Errorw("could not join register participant topic", err)
-			_ = participant.Close(true, types.ParticipantCloseReasonMessageBusFailed, false)
-			return err
-		}
+
```
Comment (boks1971) [inline pkg/service/roommanager.go:438]: Should this still be conditioned on Enabled?

File: pkg/service/roommanager.go [hunk 2]
```diff
@@ -530,10 +528,17 @@ func (r *RoomManager) getOrCreateRoom(ctx context.Context, roomName livekit.Room
...
+		killRoomServer()
+		r.lock.Unlock()
+		return nil, err
+	}
+
 	newRoom.OnClose(func() {
-		if r.config.PSRPC.Enabled {
-			r.roomServer.DeregisterAllRoomTopics(rpc.FormatRoomTopic(roomName))
-		}
+		killRoomServer()
```
Comment (boks1971) [inline pkg/service/roommanager.go:541]: Does Deregister happen in kill?

**Review by boks1971 — APPROVED** [review]

File: pkg/service/roommanager.go (hunk 1 above)
Comment (paulwe) [reply pkg/service/roommanager.go:438]: iirc this was how we rolled out signal so in 6 months or whenever we make psrpc the default existing deployments will be forward compatible

File: pkg/service/roommanager.go (hunk 2 above)
Comment (paulwe) [reply pkg/service/roommanager.go:541]: yes

File: pkg/service/roommanager.go (hunk 1 above)
Comment (boks1971) [reply pkg/service/roommanager.go:438]: Sorry about the confusion. If this is not constrained by `Enabled`, is both legacy and PsRPC handling requests?

Comment (paulwe) [reply pkg/service/roommanager.go:438]: yes

Comment (boks1971) [reply pkg/service/roommanager.go:438]: got it, thank you @paulwe

**Pull request merged**